From 95e193469e52f4faa6471652ac490148f6c65f29 Mon Sep 17 00:00:00 2001
From: jchaloup <jchaloup@redhat.com>
Date: Thu, 12 Jun 2014 10:24:27 +0200
Subject: [PATCH] pam cache turned off

---
 src/pam/pam_cgroup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pam/pam_cgroup.c b/src/pam/pam_cgroup.c
index 0618dc1..aa0bd62 100644
--- a/src/pam/pam_cgroup.c
+++ b/src/pam/pam_cgroup.c
@@ -137,7 +137,7 @@ PAM_EXTERN int pam_sm_open_session(pam_handle_t *pamh, int flags,
 	/* Note: We are using default gid here. Is there a way to determine
 	 * under what egid service will be provided?
 	 */
-	ret = cgroup_change_cgroup_uid_gid_flags(pwd->pw_uid, pwd->pw_gid, pid, CGFLAG_USECACHE);
+	ret = cgroup_change_cgroup_uid_gid_flags(pwd->pw_uid, pwd->pw_gid, pid, 0);
 	if (ret) {
 		if (ctrl & PAM_DEBUG_ARG)
 			pam_syslog(pamh, LOG_ERR, "Change of cgroup for process"
-- 
1.9.3

